workflows/rnaseq.nf

```diff
@@ -195,11 +191,10 @@ workflow RNASEQ {
     .reads
     .map {
         meta, fastq ->
-            def meta_clone = meta.clone()
-            meta_clone.id = meta_clone.id.split('_')[0..-2].join('_')
-            [ meta_clone, fastq ]
+            new_id = meta.id - ~/_T\d+/
+            [ meta + [id: new_id], fastq ]
     }
-    .groupTuple(by: [0])
+    .groupTuple()
```

Comment:
what about using `size`here so the tuple advances as soon as possible?

Reply:
Good question! I'll leave @robsyme to comment here. Asked him exactly the same question.

Reply:
To emit asap, we'd need to calculate the number of technical replicates at an earlier stage - probably during samplesheet validation. This is certainly possible, but because there are no long-running processes between samplesheet generation and this grouping operation, there is no real speedup to be gained.

Reply:
To emit asap, we'd need to calculate the number of technical replicates at an earlier stage - probably during samplesheet validation. This is certainly possible, but because there are no long-running processes between samplesheet generation and this grouping operation, there is no real speedup to be gained.

Reply:
Ok makes sense.